charts/controls/Controls.tsx

```diff
@@ -480,12 +480,13 @@ class TimelineControl extends React.Component<{ chart: ChartConfig }> {
                 />
             )
         } else if (chart.isScatter) {
-            const years = chart.scatterTransform.timelineYears
-            if (years.length === 0) return null
+            const { startYear, endYear, timelineYears } = chart.scatterTransform
+            if (timelineYears.length === 0) return null
+            chart.timeDomain = [startYear, endYear]
```

Comment:
Render methods should not mutate observables.

Reply:
Another one of those things that I wish Mobx showed us at compile time.

Reply:
agree, this would be a great compile-time resource!